### ISSUE TITLE
Autocomplete for email fields

### ISSUE THREAD

**Issue by jordanlapointe** [LeaVerou/awesomplete#8]
Many of the forms we use on the web require an email address and require that the email address is free from errors. It's a tedious process which could be made nicer with awesomplete.

To help solve this, there could be an option to only show autocomplete suggestions after a user has typed an @ character. Then, suggestions could be provided like "gmail.com" or "outlook.com". This could help cut down on common typos and save people a little bit of time.

**Comment by LeaVerou** [comment]
That’s a brilliant idea!
We should investigate how this could be written as a separate script (a plugin if you may) on top of Awesomplete. i.e. what extra hooks/settings are needed. There should be a way to specify that autocomplete happens only after a specific string has been typed and a way to intercept how selected suggestions replace the value the user has typed.

**Comment by jordanlapointe** [comment]
I love how tiny awesomplete is, so building this sort of functionality as a dependent module makes a lot of sense.

**Comment by acrogenesis** [comment]
Awesome idea!

**Comment by sillero** [comment]
I managed to solve this using the existing options. code below

``` javascript
//email example
(function(){
    var input = document.getElementById("emailInput");
    var awesomplete = new Awesomplete(input, {
        list: ["@aol.com", "@att.net", "@comcast.net", "@facebook.com", "@gmail.com", "@gmx.com", "@googlemail.com", "@google.com", "@hotmail.com", "@hotmail.co.uk", "@mac.com", "@me.com", "@mail.com", "@msn.com", "@live.com", "@sbcglobal.net", "@verizon.net", "@yahoo.com", "@yahoo.co.uk"],
        item: function(text, input){
            var newText = input.slice(0, input.indexOf("@")) + text;

            return Awesomplete.$.create("li", {
                innerHTML: newText.replace(RegExp(input.trim(), "gi"), "<mark>$&</mark>"),
                "aria-selected": "false"
            });
        },
        filter: function(text, input){
            var atPosition = input.indexOf("@");
            var inputIsEmail = !!~atPosition && (input.length > atPosition + 1);
            var emailPart = input.slice(input.indexOf("@"));
            var textMatchWithInput = RegExp("^" + emailPart.trim(), "i").test(text);

            return (inputIsEmail && textMatchWithInput);
        }
    });
})();
```

a more complete list of email domains you can find here https://github.com/mailcheck/mailcheck/wiki/list-of-popular-domains

**Comment by LeaVerou** [comment]
This is very cool @sillero!! Is there a demo somewhere? I’m failing to see how it would work when the user selects something, but I haven’t tried running it. 
If it works, I’d love to clean up the code and add it as an example on the page. E.g. we could have an "Advanced examples" section, with this and e.g. using it with Ajax etc.

**Comment by sillero** [comment]
Hey @LeaVerou, I forked and created an advanced examples section with the email auto-complete. You can check it out here http://sillero.github.io/awesomplete/#advanced-examples

I'll make more examples, ajax sounds cool. I'll go through the issues and see if another use case pops up.

**Comment by jordanlapointe** [comment]
That is pretty slick. Thanks for writing it!

**Comment by LeaVerou** [comment]
This is very cool. Could you send a PR @sillero? I wouldn't it put it second though, I would put it at the end.

**Comment by sillero** [comment]
Done, PR on the list. Anything else just drop a comment :)

**Comment by vlazar** [comment]
Closing this, as the email example is now in the docs. It landed in #6614.
